src/term.c

```diff
@@ -4554,6 +4554,7 @@ check_termcode(
 			    /* Skip over the digits, the final char must
 			     * follow. */
 			    for (j = slen - 2; j < len && (isdigit(tp[j])
+							 || tp[j] == '-'
```

Comment:
There are tests for mouse terminal code. See src/testdir/test_termcodes.vim
At the moment it only tests 'xterm2' and 'sgr' mouse, but we could add test for 'urxvt'.